src/server/server.ts

```diff
@@ -59,6 +59,52 @@ export const createServer = (
     res.status(200).json({ ok: true });
   });
 
+  app.post("/:userPoolId/oauth2/token", (req, res) => {
+    let rawBody = "";
+    req.setEncoding("utf8");
+    req.on("data", function (chunk) {
+      rawBody += chunk;
+    });
+    req.on("end", function () {
+      const target = "GetToken";
+      const route = router(target);
+      const auth = req.get("Authorization");
+      if (auth && auth.startsWith("Basic ")) {
+        const sliced = auth.slice("Basic ".length);
+        const buff = new Buffer(sliced, "base64");
+        const decoded = buff.toString("ascii");
+        const creds = decoded.split(":");
+        if (creds.length == 2) {
+          const id = creds[0];
+          const secret = creds[1];
+          rawBody += `&client_id=${id}`;
+          rawBody += `&client_secret=${secret}`;
+        }
+      }
```

Comment:
This is a bit gnarly. 😄 

I'm assuming the POST body is form encoded? I'd suggest you either:

1. Configure the [urlencoded body parser](https://expressjs.com/en/resources/middleware/body-parser.html#bodyparserurlencodedoptions) and that'll make `req.body` be key-value pairs if the correct `Content-Type` is sent on the request, then you'll have a proper object to deal with instead of passing strings around.
2. Or at least use the [raw body parser](https://expressjs.com/en/resources/middleware/body-parser.html#bodyparserrawoptions) which'll make `req.body` into a Buffer you can call toString on, instead of having to concat from a stream.

My preference would be for (1) and then also update the types of `GetToken` to something more specific too.

Reply:
Ok - I adjusted a little, lifted the URL parsing up to the top here, brought it into its own type, then injected the Authorization headers into the new request type instead of smashing it onto the string there.

I might have gone a little overboard with the type things, but I dont write a ton of typescript so got to have a little fun with it :)